lib/config_servers.js

```diff
@@ -194,13 +198,20 @@ class ConfigServers extends EventEmitter {
 
     return new Promise(function(resolve, reject) {
       co(function*() {
-        var secondaries = [];
+        // Call ismaster on all of the managers first (in parallel)
+        var ismasters = yield self.managers.map(function(server) {
```

Comment:
This is actually a reduction:
```
const secondaries = Promise.map(self.managers, server => server.ismaster()).reduce(
  (result, ismaster, idx) => {
    if (ismaster.secondary) result.push(self.managers[idx]);
    return result;
  },
  []
);
```

Reply:
Thank you! I drove myself nuts trying to find a *clean* way of doing it. I didn't know about that, very clever way of handling it.